src/amp-story-player/amp-story-player-impl.js

```diff
@@ -393,6 +431,144 @@ export class AmpStoryPlayer {
+      }
+
+      this.updateUpNextInfo_(card, title, author);
+      this.startUpNext_(container, text);
+    });
+
+    this.element_.addEventListener('navigation', () => {
+      this.endUpNext_(container);
+    });
+  }
+
+  /**
+   * Updates the up next elements with the next story's values.
+   * @param {Element} card
+   * @param {Element} title
+   * @param {Element} author
+   */
+  updateUpNextInfo_(card, title, author) {
+    const story = this.stories_[this.currentIdx_ + 1];
+    card.src = story.getAttribute('data-poster-portrait-src');
+    title.textContent = story.querySelector('span').textContent;
+    author.textContent = 'Placeholder';
+  }
+
+  /**
+   * Starts the countdown, displays and updates the up next elements.
+   * @param {Element} container
+   * @param {Element} text
+   * @private
+   */
+  startUpNext_(container, text) {
+    let counter = 10;
+    text.textContent = 'UP NEXT in ' + counter;
+    container.classList.remove('story-player-hide');
+
+    this.interval_ = setInterval(
+      function (next) {
+        counter -= 1;
+
+        text.textContent = 'UP NEXT in ' + counter;
+
+        if (counter === 0) {
+          next();
+        }
+      },
+      1000,
+      this.next_.bind(this)
+    );
+  }
+
+  /**
+   * Clears the countdown and hides the up next elements.
+   * @param {Element} container
+   * @private
+   */
+  endUpNext_(container) {
+    clearInterval(this.interval_);
+    container.classList.add('story-player-hide');
+  }
+
+  /**
+   * Creates all elements necessary for up next.
+   * @return {Object<string, Element>}
+   * @private
+   */
+  createUpNextElement_() {
+    const container = this.doc_.createElement('div');
+    container.classList.add('story-player-up-next-container');
+    container.classList.add('story-player-hide');
+
+    const card = this.doc_.createElement('img');
+    card.classList.add('story-player-up-next-card');
+
+    const cancel = this.doc_.createElement('button');
+    cancel.addEventListener('click', () => {
+      this.endUpNext_(container);
+    });
+    cancel.textContent = 'Cancel';
+    cancel.classList.add('story-player-up-next-cancel-button');
+
+    const textContainer = this.doc_.createElement('div');
+    const textElements = this.createUpNextTexts_(textContainer);
+
+    container.appendChild(card);
+    container.appendChild(cancel);
+    container.appendChild(textContainer);
+    this.rootEl_.appendChild(container);
+
+    return {container, card, ...textElements};
+  }
+
+  /**
+   * Creates up next text elements: timer, title, and author and appends
+   * those elements to the container provided.
+   * @param {!Element} textContainer
+   * @return {Object<string, Element>}
+   * @private
+   */
+  createUpNextTexts_(textContainer) {
+    textContainer.classList.add('story-player-up-next-text-container');
+
+    const text = this.doc_.createElement('p');
+    text.classList.add('story-player-up-next-text');
+    text.textContent = 'UP NEXT in 10';
```

Comment:
We'll need to localize these strings. I wonder if we can use the `LocalizationService` from `amp-story` for this.

https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/amp-story-localization-service.js#L28